app/models/compliance.rb

```diff
@@ -66,13 +66,10 @@ def self.check_compliance(target, _inputs = {})
                                                                                          :number     => id}
       end
     end
-    target_class = target.class.base_model.name.downcase
-    target_class = "vm" if target_class.match("template")
-
-    unless target.respond_to?(:compliances)
+    unless target.supports?(:compliances)
```

Comment:
I am reacting to the responds_to (adam has rubbed off on me here)

but the old way asking if the relation is defined is probably fine.
looking for suggestions on this one.

Reply:
This feels like the wrong change, because you've changed the way supports is defined.  Before it was either supported or not supported.  Now, you've introduced "sometimes supported" via `has_compliance_policies?`.

I think this needs to change further such that the raise on the next line takes into account the `unsupported_reason`.  Either that or this should be `target.class.supports?(:compliances)`.

@agrare Keep me honest here :)

Reply:
Right if you just want to know if the class supports it (aka bypass the block) then you can call `target.class.supports?(:compliances)`

Reply:
nit/edit: call `target.class.support?(:check_compliance)`

that was such a good idea. thnx